libcloud/dns/drivers/godaddy.py

```diff
@@ -0,0 +1,521 @@
+                     Note: If you want to create a record for a base domain
+                     name, you should specify empty string ('') for this
+                     argument.
+        :type  name: ``str``
+
+        :param zone: Zone where the requested record is created.
+        :type  zone: :class:`Zone`
+
+        :param type: DNS record type (A, AAAA, ...).
+        :type  type: :class:`RecordType`
+
+        :param data: Data for the record (depends on the record type).
+        :type  data: ``str``
+
+        :param extra: Extra attributes (driver specific). (optional)
+        :type extra: ``dict``
+
+        :rtype: :class:`Record`
+        """
+        if extra is None:
+            extra = {}
+        new_record = {}
+        if type is RecordType.SRV:
+            new_record = {
+                'type': type,
+                'name': name,
+                'data': data,
+                'priority': 1,
+                'ttl': extra['ttl'] if hasattr(extra, 'ttl') else 5,
+                'service': extra['service']
+                if hasattr(extra, 'service') else '',
+                'protocol': extra['protocol']
+                if hasattr(extra, 'protocol') else '',
+                'port': extra['port']
+                if hasattr(extra, 'port') else '',
+                'weight': extra['weight']
+                if hasattr(extra, 'weight') else ''
+            }
+        else:
+            new_record = {
+                'type': type,
+                'name': name,
+                'data': data,
+                'priority': 1,
+                'ttl': extra['ttl'] if hasattr(extra, 'ttl') else 5
+            }
+        self.connection.request(
+            '/v1/domains/{0}/records'.format(
+                zone.domain), method='PATCH',
+            data=[new_record])
+        id = '{0}:{1}'.format(name, type)
+        return Record(
+            id=id, name=name,
+            type=type, data=data,
+            zone=zone, driver=self,
+            extra=extra)
+
+    def update_record(self, record, name, type, data, extra=None):
+        """
+        Update an existing record.
+
+        :param record: Record to update.
+        :type  record: :class:`Record`
+
+        :param name: Record name without the domain name (e.g. www).
+                     Note: If you want to create a record for a base domain
+                     name, you should specify empty string ('') for this
+                     argument.
+        :type  name: ``str``
+
+        :param type: DNS record type (A, AAAA, ...).
+        :type  type: :class:`RecordType`
+
+        :param data: Data for the record (depends on the record type).
+        :type  data: ``str``
+
+        :param extra: (optional) Extra attributes (driver specific).
+        :type  extra: ``dict``
+
+        :rtype: :class:`Record`
+        """
+        if extra is None:
+            extra = {}
+        new_record = {}
+        if type is RecordType.SRV:
```

Comment:
It seems there is a lot of duplication code between this method and create record - would be good to refactor common stuff in a utility method.